lightning/src/util/errors.rs

```diff
@@ -37,7 +37,7 @@ pub enum APIError {
 	/// too-many-hops, etc).
 	InvalidRoute {
 		/// A human-readable error message
-		err: &'static str
+		err: String
```

Comment:
If we're going to add serialization, should we just make this an enum? Seems wasteful to serialize static strings.

Reply:
Eh, seems a bit unrelated to the PR, there'd be about 7 variants and they should all basically never happen..